autogpt_platform/backend/backend/blocks/test/test_llm.py

```
@@ -1663,3 +1663,89 @@ async def test_available_tokens_shrink_for_claude_5(self):
+        call_provider.assert_awaited_once()
+        assert call_provider.call_args.kwargs["api_key"] == ""
+        assert call_provider.call_args.kwargs["provider"] == llm.ProviderName.OLLAMA
+
+    @pytest.mark.asyncio
+    async def test_remote_model_without_credentials_fails_before_provider_call(self):
+        with patch(
+            "backend.util.llm.providers.call_provider", new_callable=AsyncMock
+        ) as call_provider:
+            with pytest.raises(
+                ValueError, match=r"Credentials are required for openai/"
+            ):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)